CMakePresets.json

```diff
@@ -101,6 +101,17 @@
       "description": "Visual Studio-based x64 build with vcpkg dependencies.",
       "architecture": "x64"
     },
+    {
+      "name": "macOS",
```

Comment:
This change seems unrelated to the purpose of the PR, but OK

Reply:
Oops, I meant to stash that, but it's honestly probably fine to leave in. I needed a preset that used Apple Clang and Homebrew libraries... there wasn't one already, so I added it.